### PR TITLE
update max user agent length.

### PR DESCRIPTION
**Pull request by ShivangiReja**
## Description

Brief description of the changes made in the PR. This helps in making better change log
- There's a hard limit of 128 characters on the user agent string and it could easily go over the limit( custom user agent string etc).
- This pr updates the limit to 512 characters.

# Reference to any github issues
-  https://github.com/Azure/azure-sdk-for-js/issues/1211

### CODE REVIEW

File: lib/ConnectionContextBase.ts [hunk 1]
```diff
@@ -140,7 +140,7 @@ export module ConnectionContextBase {
     const userAgent = parameters.connectionProperties.userAgent;
     if (userAgent.length > Constants.maxUserAgentLength) {
       throw new Error(
-        `The user-agent string cannot be more than 128 characters in length.` +
+        `The user-agent string cannot be more than 512 characters in length.` +
```
Comment (ramya-rao-a) [inline lib/ConnectionContextBase.ts]: Lets use the constant `maxUserAgentLength` here instead of hard coding 512.

**Comment by ShivangiReja** [comment]
@ramya-rao-a As discussed offline we will add a log for all the connection properties after successfully creating a connection in EventHubs.

**Comment by ramya-rao-a** [comment]
@ShivangiReja https://github.com/Azure/azure-sdk-for-js/pull/2270 was the fix done in the main repo to fix the auditing issue.
Please do the same here to get the CI check

**Comment by mikeharder** [comment]
@ramya-rao-a , @ShivangiReja : I will submit a separate PR to update nyc to version 14.0.0.

**Pull request merged**